### ISSUE TITLE
[a11y] Spin button is missing semantics role

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#161558]
When using this widget https://api.flutter.dev/flutter/material/showDatePicker.html, the next and previous arrow button semantics role should have the following 

![Image](https://github.com/user-attachments/assets/fea36e9d-4875-470a-883f-e46df3656f73)

| OS | role |
|--------|--------|
| web | spinbutton |
| ios | - |
| macos | NSAccessibilityIncrementorRole |
| windows | ROLE_SYSTEM_SPINBUTTON |
| android | BottomSheetDragHandleView |
